lib/sqlite.core.js

```diff
@@ -588,39 +588,42 @@ SQLitePluginTransaction.prototype.run = function() {
       }
     };
   };
+  
   i = 0;
-  mycbmap = {};
+  mycbmap = [];
   while (i < batchExecutes.length) {
     request = batchExecutes[i];
-    mycbmap[i] = {
+    mycbmap.push({
       success: handlerFor(i, true),
       error: handlerFor(i, false)
-    };
+    });
     tropts.push({
       qid: 1111,
       sql: request.sql,
       params: request.params
     });
     i++;
   }
+
   let mysuccess = function(result) {
-    var j, last, q, r, ref, res, type;
+    var j, last, q, r, res, type;
     if (result.length == 0){
       return;
     }
     last = result.length - 1;
-    for (i = j = 0, ref = last; 0 <= ref ? j <= ref : j >= ref; i = 0 <= ref ? ++j : --j) {
```

Comment:
Honestly, I don't understand the reason behind such complexity.
Though I'm not a js developer so I might be missing something.

for (i = j = 0, ref = last; 0 <= ref ? j <= ref : j >= ref; i = 0 <= ref ? ++j : --j) {

the ref could be == -1 only in case the map is empty.
I don't think we need to handle this case there.

In all other cases, the ref is >= 0.
So 0 <= ref is always true and we never will use --j and  j >= ref.